### PR TITLE
SparseML dependency on `compressed-tensors`

### PR DESCRIPTION
**Pull request by dbogunowicz**
GHA will be green after: https://github.com/neuralmagic/compressed-tensors/pulls lands

### CODE REVIEW

File: setup.py
Comment (bfineran) [inline setup.py:58]: you can't build/publish a wheel with a dependency with git deps due to privacy restrictions

**Review by bfineran — CHANGES_REQUESTED** [review]
biggest thing I'd like to talk through with you and @Satrat is moving the actual configs into sparsetensors. specifically having configs take in the sparsification structure as an input so we don't need to couple the analysis code with the config classes

**Comment by Satrat** [comment]
@bfineran @dbogunowicz is there anything blocking this from being merged??

**Comment by dbogunowicz** [comment]
@Satrat I'd say the readiness of the pypi package. Now since we have the basic pathways in place, I will get this PR and its counterpart in vLLM in order. But let's discuss the readiness status today.

**Review by rahul-tuli — CHANGES_REQUESTED** [review]

File: tests/sparseml/transformers/test_clear_ml.py
Comment (dbarbuzzi) [inline tests/sparseml/transformers/test_clear_ml.py:25]: Why not something like this, since the whole file appears to be for testing clearml — then you wouldn't need any special skip decorators on the test(s).

```python
try:
    from clearml import Task
except ImportError:
    pytest.skip("clearml not installed", allow_module_level=True)
```

Alternatively, [`pytest.importorskip`](https://docs.pytest.org/en/8.0.x/reference/reference.html#pytest-importorskip-ref) works at a module-level as well:

```python
clearml = pytest.importorskip(clearml, reason="clearml module is required")
# use clearml.Task in later usage, or do Task = clearml.Task after imports to put `Task` in the namespace directly
```

Comment (dbogunowicz) [reply tests/sparseml/transformers/test_clear_ml.py:25]: awesome, thanks!

Comment (dbogunowicz) [reply tests/sparseml/transformers/test_clear_ml.py:25]: those solutions do not work ad hoc for me. let's just get this in; i really would like to unblock team with this PR

Comment (dbarbuzzi) [reply tests/sparseml/transformers/test_clear_ml.py:25]: Strange, I’ve not had issues using either approach. Whatever works, though!

**Review by bfineran — DISMISSED** [review]

**Review by Satrat — APPROVED** [review]

**Review by Satrat — APPROVED** [review]

**Review by rahul-tuli — APPROVED** [review]

**Review by bfineran — APPROVED** [review]

**Pull request merged**
